keps/sig-network/2004-topology-aware-routing/README.md

```diff
@@ -0,0 +1,578 @@
+  - [Test Plan](#test-plan)
+  - [Observability](#observability)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes clusters are increasingly deployed in multi-zone environments but
+unfortunately network routing has not caught up with that. This KEP proposes an
+automatic topology aware routing mechanism with a new Service annotation. This
+new annotation will provide three options of different routing options.
+Currently our proposal will only focus on topology aware routing at zone level.
+
+## Motivation
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+Network traffic is routed randomly to any endpoint matching a Service. Some
+users might want the traffic to stay in the same zone for the following
+reasons:
+  - Cost savings: Keeping traffic within a zone can limit cross-zone networking
+costs.
+  - Performance: Traffic within a zone usually has less latency and bandwidth
+constraints, having a better performance than traffic leaving the zone.
+
+In this KEP we are going to focus on avoiding cross-zone traffic when in-zone
+endpoints would suffice. We're attempting to provide a simple and more
+automatic approach to topology aware routing. This API will still allow users
+to require traffic to stay within a zone, or simply to indicate that they
+prefer to keep traffic in the same zone if there's sufficient capacity. With
+this approach no new API fields are required, we can accomplish this simply
+with a new annotation on Service.
+
+
+### Goals
+- Provide a simple way for users to indicate their preference for keeping
+traffic in zone.
+- Provide different approaches to users who have a strong desire over the
+routing policy:
+  - Balanced: Use the original approach to ensure traffic is distributed across
```

Comment:
Updated, thanks!

Reply:
I'm still not sure "Balanced" vs "Spread" vs "Random"

Reply:
I like Random/Randomized because it carries the iptables extension name used by kube-proxy, kind of tribute :smile:

Reply:
Updated to random for now since it seems to be the most common suggestion. Open to other ideas, especially if we end up changing the name of the field/annotation.